htdocs/api/class/api_setup.class.php

```diff
@@ -1341,17 +1343,15 @@ public function postExtrafields($attrname, $elementtype, $request_data = null)
 			$extrafields->$field = $this->_checkValForAPI($field, $value, $extrafields);
 		}
 
+		$entity = getEntity('extrafields');
```

Comment:
getEntity can be used to return the list of allowed entities. It can be '1,4,9'
Here you need the entity where the post is addressed. You are right, providing it as parameter is not enough.
I think you can try use $conf->entity
to get the entity the code is running into (solution 1) but a http header can be used in the POST to work with another value. Value is checked before setting it into conf->entity.
Another solution (solution 2) may be to allow parameter entity and check that value is among the list of values returned by getEntity.
Sol 1 looks better. Thanks to try it.

Reply:
`$entity = DolibarrApiAccess::$user->entity;`

found in api_documents.class.php seems to work.

Reply:
except that database contents are different than those edited inside Dolibarr

inside Dolibarr is created with 1 as entity, where as using the line above uses 0 for entity :-( and $conf->entity did not work

Reply:
https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/api/class/api_documents.class.php#L677-L680

works